modules/tinymce/src/plugins/table/test/ts/browser/command/ModifyClassesCommandsTest.ts

```diff
@@ -0,0 +1,249 @@
+      context('When there is no class already on the cell', () => {
+        const getContentWithoutClass = () => {
+          return (
+            '<table>' +
+              '<tbody>' +
+                '<tr>' +
+                  '<td>1</td>' +
+                  '<td>2</td>' +
+                '</tr>' +
+              '</tbody>' +
+            '</table>'
+          );
+        };
+
+        const getContentWithClass = () => {
+          return (
+            '<table class=" a">' +
+              '<tbody>' +
+                '<tr>' +
+                  '<td>1</td>' +
+                  '<td>2</td>' +
+                '</tr>' +
+              '</tbody>' +
+            '</table>'
+          );
+        };
+
+        it('Can be toggled on', () => {
+          const editor = hook.editor();
+          editor.setContent(getContentWithoutClass());
+
+          TinySelections.setSelection(editor, [ 0, 0, 0 ], 0, [ 0, 0, 0 ], 1, true);
+
+          execCmdAndAssertEvent('TINY-7163', editor, 'mceTableToggleClass', false, 'a');
+
+          assert.equal(cleanTableHtml(editor.getContent()), getContentWithClass());
+        });
+
+        it('Can be toggled off', () => {
+          const editor = hook.editor();
+          editor.setContent(getContentWithClass());
+
+          TinySelections.setSelection(editor, [ 0, 0, 0 ], 0, [ 0, 0, 0 ], 1, true);
+
+          execCmdAndAssertEvent('TINY-7163', editor, 'mceTableToggleClass', false, 'a');
+
+          assert.equal(cleanTableHtml(editor.getContent()), getContentWithoutClass());
+        });
+      });
+
+      context('When there is already a class on the cell, which should not be deleted', () => {
+        const getContentWithoutClass = () => {
+          return (
+            '<table class="b">' +
+              '<tbody>' +
+                '<tr>' +
+                  '<td>1</td>' +
+                  '<td>2</td>' +
+                '</tr>' +
+              '</tbody>' +
+            '</table>'
+          );
+        };
+
+        const getContentWithClass = () => {
+          return (
+            '<table class="b a">' +
+              '<tbody>' +
+                '<tr>' +
+                  '<td>1</td>' +
+                  '<td>2</td>' +
+                '</tr>' +
+              '</tbody>' +
+            '</table>'
+          );
+        };
+
+        it('Can be toggled on', () => {
+          const editor = hook.editor();
+          editor.setContent(getContentWithoutClass());
+
+          TinySelections.setSelection(editor, [ 0, 0, 0 ], 0, [ 0, 0, 0 ], 1, true);
+
+          execCmdAndAssertEvent('TINY-7163', editor, 'mceTableToggleClass', false, 'a');
+
+          assert.equal(cleanTableHtml(editor.getContent()), getContentWithClass());
+        });
+
+        it('Can be toggled off', () => {
+          const editor = hook.editor();
+          editor.setContent(getContentWithClass());
+
+          TinySelections.setSelection(editor, [ 0, 0, 0 ], 0, [ 0, 0, 0 ], 1, true);
```

Comment:
Can you use some different selections where it makes sense throughout this test please? That makes sure it works for things that aren't just in the first cell.